### PR TITLE
Vet Clinic #2 

### PR DESCRIPTION
**Pull request by wassimchakib**
## What

Solving exercises from [Microverse assignment](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md)

### Changes details

- Added transactions for each exercise.
- Added queries

### Solutions

 Inside a transaction:
  - Update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
  - Update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.
<details>
 <summary></summary>

</details>

- Now, take a deep breath and... **Inside a transaction** delete all records in the `animals` table, then roll back the transaction.
- After the rollback verify if all records in the `animals` table still exists. After that, you can start breathing as usual ;) 

<details>
 <summary></summary>

</details>

- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction

<details>
 <summary></summary>

</details>

 - **_How many animals are there?_**

<details>
 <summary></summary>

</details>

- **_How many animals have never tried to escape?_**

<details>
 <summary></summary>

</details>

- **_What is the average weight of animals?_**

<details>
 <summary></summary>

</details>

- **_Who escapes the most, neutered or not neutered animals?_**

<details>
 <summary></summary>

</details>

- **_What is the minimum and maximum weight of each type of animal?_**

<details>
 <summary></summary>

</details>

- **_What is the average number of escape attempts per animal type of those born between 1990 and 2000?_**

<details>
 <summary></summary>

</details>

## Missed anything?

- [x] Explained the purpose of this PR.
- [x] Self reviewed the PR.
- [ ] Added or updated test cases.
- [ ] Informed of breaking changes, testing and migrations (if applicable).
- [ ] Updated documentation (if applicable).
- [x] Attached screenshots (if applicable).

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -14,4 +14,44 @@ select * from animals where neutered = '1';
...
 /* Find all animals with a weight between 10.4kg and 17.3kg 
 (including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
-select * from animals where weight_kg between 10.4 and 17.3;
\ No newline at end of file
+select * from animals where weight_kg between 10.4 and 17.3;
+
+/* Transaction 1 : update the animals table by setting the species column to unspecified.*/
+begin;
+Update animals set species = 'unspecified';
+rollback;
```
Comment (omar-muhamad) [inline queries.sql:22]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction), Please Verify that change was made and verify that the species columns went back to the state before the transaction.

File: queries.sql [hunk 2]
```diff
@@ -14,4 +14,44 @@ select * from animals where neutered = '1';
...
+
+/* Transaction 2 */
+begin;
+Update animals set species = 'digimon' where name like '%mon';
+Update animals set species = 'pokemon' where species is null;
+commit;
```
Comment (omar-muhamad) [inline queries.sql:29]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Verify%20that%20change%20was%20made%20and%20persists%20after%20commit.), Please Verify that change was made and persists after commit.

File: queries.sql [hunk 3]
```diff
@@ -14,4 +14,44 @@ select * from animals where neutered = '1';
...
+
+/* Transaction 3 */
+begin;
+delete from animals;
+rollback;
```
Comment (omar-muhamad) [inline queries.sql:35]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=After%20the%20rollback%20verify%20if%20all%20records%20in%20the%20animals%20table%20still%20exists), Please verify if all records in the animals table still exist.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @wassimchakib,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ ✔️ ✔️ 

Hi @wassimchakib, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**